### PR TITLE
[llvmonly] Handle bitcode exception object manipulation racing

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

File: mono/mini/mini-exceptions.c [hunk 1]
```diff
@@ -3044,7 +3044,7 @@ throw_exception (MonoObject *ex, gboolean rethrow)
 		mono_ex = (MonoException*)ex;
 
 	// Note: Not pinned
```
Comment (evincarofautumn) [inline mono/mini/mini-exceptions.c]: This comment is wrong now.

Comment (alexanderkyte) [inline mono/mini/mini-exceptions.c]: Good catch

**Comment by alexanderkyte** [comment]
I don't think this is actually the right fix, but it's one that works. I believe that this relates to the issue we saw before about not scanning SIMD registers. If we assign `x = mono_ex->trace_ips` then we can use `x` without it breaking. 

For specifics, I'm watching this break on ARM with bitcode. The `bitcode_arm_linux` branch, excluding the fix related to this, can reproduce deterministically when trying to run the corlib suite.

**Comment by vargaz** [comment]
The change looks ok, but the failure should not happen.

**Comment by alexanderkyte** [comment]
Yeah, it was a weird one. I think it might be related to register scanning

**Pull request closed without merging**